### ISSUE TITLE
Confusing error message when both an attr descriptor and a property are defined w/ the same name

### ISSUE THREAD

**Issue by cournape** [python-attrs/attrs#30]
Not sure that's really fixeable, but it took me a while to detect this bug in my code:

```
from attr import attributes, attr

@attributes
class Foo(object):
    a = attr()
    @property
    def a(self):
        return "yolo"

Foo("value")
```

raises

```
Traceback (most recent call last):
  File "failure.py", line 13, in <module>
    Foo("value")
TypeError: __init__() takes exactly 1 argument (2 given)
```

**Comment by hynek** [comment]
¯\_(ツ)_/¯

No idea what to do save starting an FAQ…

**Comment by infinite-Joy** [comment]
```
class SeriousCoordinates(object):

	def __init__(self, x):
		self.x = x
	
	@property
	def x(self):
		return "sdkjh"

print(SeriousCoordinates(1))
```
gives
```
Traceback (most recent call last):
  File "issue30_001.py", line 10, in <module>
    print(SeriousCoordinates(1))
  File "issue30_001.py", line 4, in __init__
    self.x = x
AttributeError: can't set attribute
```
Maybe here too we can have something similar.

**Comment by mvalkon** [comment]
@infinite-Joy this is standard python behavior and not related to `attrs` at all. The `@property` decorator is typically used to provide a mechanism for hiding an internal field, and providing a _getters_ and _setters_ to dictate what can and cannot be done with said field. The method name decorated with `@property` becomes a field name (in your case, `SeriousCoordinates.x`, which already exists in `__init__`). Field names generated by the `@property`-decorator cannot be assigned to directly, which is why an `AttributeError` is raised.

**Comment by hynek** [comment]
fixed in #370
